pkg/query/api.go

```diff
@@ -0,0 +1,445 @@
+package query
```

Comment:
Probably good to have some sort of notice on this file that this is largely copied from prometheus/prometheus and the respective licensing applies. License header needs to be included I think.

Reply:
I added that notice to the API struct comment, but can do that below header